docs/painless/painless-guide/painless-execute-script.asciidoc

```diff
@@ -154,13 +249,560 @@ POST /_scripts/painless/_execute
+
+[source,console]
+----
+PUT /my-index-000001
+{
+  "mappings": {
+    "properties": {
+      "name": {
+        "type": "keyword"
+      },
+      "author": {
+        "type": "keyword"
+      },
+      "release_date": {
+        "type": "date"
+      },
+      "page_count": {
+        "type": "double"
+      }
+    }
+  }
+}
+----
+
+You can then write a script in the `boolean_field` context that indicates
+whether a book was published before the year 1972:
+
+[source,console]
+----
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": """
+       emit(doc['release_date'].value.dayOfYear < 1972);
+    """
+  },
+  "context": "boolean_field",
+  "context_setup": {
+    "index": "my-index-000001",
+    "document": {
+      "name": "Dune",
+      "author": "Frank Herbert",
+      "release_date": "1965-06-01",
+      "page_count": 604
+    }
+  }
+}
+----
+// TEST[continued]
+
+Because _Dune_ was published in 1965, the result returns as `true`:
+
+[source,console-result]
+----
+{
+  "result" : [
+    true
+  ]
+}
+----
+
+Similarly, you could write a script that determines whether the first name of
+an author exceeds a certain number of characters. The following script operates
+on the `author` field to determine whether the author's first name contains at
+least one character, but is less than five characters:
+
+[source,console]
+----
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": """
+       int space = doc['author'].value.indexOf(' ');
+       emit(space > 0 && space < 5);
+    """
+  },
+  "context": "boolean_field",
+  "context_setup": {
+    "index": "my-index-000001",
+    "document": {
+      "name": "Dune",
+      "author": "Frank Herbert",
+      "release_date": "1965-06-01",
+      "page_count": 604
+    }
+  }
+}
+----
+// TEST[continued]
+
+Because `Frank` is five characters, the response returns `false` for the script
+valuation:
+
+[source,console-result]
+----
+{
+  "result" : [
+    false
+  ]
+}
+----
+
+[[painless-runtime-datetime]]
+===== `date_time`
+Several options are available for using
+<<painless-datetime,using datetime in Painless>>. In this example, you'll
+estimate when a particular author starting writing a book based on its release
+date and the writing speed of that author. The example makes some assumptions,
+but shows to write a script that operates on a date while incorporating
+additional information.
+
+Add the following fields to your index mapping to get started:
+
+[source,console]
+----
+PUT /my-index-000001
+{
+  "mappings": {
+    "properties": {
+      "name": {
+        "type": "keyword"
+      },
+      "author": {
+        "type": "keyword"
+      },
+      "release_date": {
+        "type": "date"
+      },
+      "page_count": {
+        "type": "long"
+      }
+    }
+  }
+}
+----
+
+The following script adds a variable named `pageTime`. The value for this field
+is `28800000` milliseconds, which equates to eight hours. We'll assume that this
```

Comment:
I'd say what you think the script does, specifically it makes a terrible guess about when the author started writing the book based solely on the page count and some bonkers assumptions about how quickly they write. Once you've said something about that you can get how it works. In this case I might even intentionally come across a bit snarky because I think the example is a bit silly, but its *contained* which is what is important about it.

This script is far to hacky to "assess" anything. Its meant to be silly.

Reply:
Also, for what it is worth, I think it'd be a little easier to read if we commented the script. Maybe:
```
    String author = doc['author'].value;
    long pageTime = 28800000;  // eight hours
    if (author == 'Robert A. Heinlein') {
      pageTime /= 2; // he's obviously fast
    } else if (author == 'Alastair Reynolds') {
      pageTime *= 2; // space opera is slower
    }
    emit(doc['release_date'].value.toInstant().toEpochMilli() - pageTime * doc['page_count'].value);
```

If you do something like this then there isn't really much need to explain the "how" the script works in prose - just what you think it does.

Reply:
Good callouts, and thanks for reminding me that we can be fun in our docs 🥳